telemetry-core/src/main/java/com/newrelic/telemetry/util/IngestWarnings.java

```diff
@@ -0,0 +1,91 @@
+package com.newrelic.telemetry.util;
+
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IngestWarnings {
+  private static final Logger logger = LoggerFactory.getLogger(IngestWarnings.class);
+  private static final int MAX_NUMBER_OF_EVENT_ATTRIBUTES =
+      254; // Public documentation says 255, but true maximum is 254
+  private static final int MAX_NUMBER_OF_METRIC_ATTRIBUTES = 100;
+  private static final int MAX_NUMBER_OF_LOG_ATTRIBUTES =
+      254; // Public documentation says 255, but true maximum is 254
+
+  private static final int MAX_ATTRIBUTE_NAME_LENGTH = 255;
+  private static final int MAX_ATTRIBUTE_VALUE_LENGTH = 4096;
+
+  public void eventWarningNumAttributes() {
+    logger.warn(
+        "The number of attributes in this event is greater than the maximum allowed attributes per event.");
+  }
+
+  public void metricWarningNumAttributes() {
+    logger.warn(
+        "The number of attributes in this metric is greater than the maximum allowed attributes per metric.");
+  }
+
+  public void logWarningNumAttributes() {
+    logger.warn(
+        "The number of attributes in this log is greater than the maximum allowed attributes per log.");
+  }
+
+  public void attributeNameWarning(String attributeName) {
+    logger.warn(
+        "The length of the attribute named "
```

Comment:
For these log messages, I would like for you to look into using string formatting. It'll look similar to this pattern in the batchDataSender.  If you use the type information you'll be getting from the class (see comment. below), you can also remove the word log - metric - event from the message and replace with {}.  

```
      logger.debug("Sending json for {} : {} ", batchType, json);
```